documenti WIF support in fusion (#9439)
fishtown-analytics/docs.getdbt.com: website/docs/docs/local/connect-data-platform/bigquery-setup.md

</TabItem>

<TabItem value="Workload Identity Federation (Microsoft Entra)">

Workload Identity Federation (WIF) lets you authenticate to BigQuery using credentials issued by an external OAuth identity provider &mdash; currently Microsoft Entra &mdash; without managing a Google service account key file. 

> <Constant name="fusion" /> exchanges an Entra-issued token for short-lived Google credentials through a BigQuery [workload identity pool](https://cloud.google.com/iam/docs/workload-identity-federation).

Reviewer: a workload identity pool is a Google Cloud / IAM construct, not BigQuery-specific.

```suggestion
<Constant name="fusion" /> exchanges an Entra-issued token for short-lived Google credentials through a Google Cloud [workload identity pool](https://cloud.google.com/iam/docs/workload-identity-federation).
```

> <Constant name="fusion" /> exchanges an Entra-issued token for short-lived Google credentials through a Google Cloud [workload identity pool](https://cloud.google.com/iam/docs/workload-identity-federation).

Reviewer 2: [Non-blocker] Should we link to this instead?
```suggestion
<Constant name="fusion" /> exchanges an Entra-issued token for short-lived Google credentials through a Google Cloud [workload identity pool](https://docs.cloud.google.com/iam/docs/workload-identity-federation#pools).
```

Before selecting this method, an administrator must configure a workload identity pool and provider in GCP that trusts your Entra tenant. You'll need the resulting workload pool provider path.

> #### Required fields

Reviewer: May be call this a `Configuration Fields` since we have included optional field in this table

Author: good callout!

> #### Required fields

Author: ```suggestion
#### Configuration fields
```

> Here are the required fields for the WIF config in your `profiles.yml` file:

Author: ```suggestion
Here are the required or optional fields for the WIF config in your `profiles.yml` file:
```

> Here are the required or optional fields for the WIF config in your `profiles.yml` file:

Reviewer 2: ```suggestion
Here are the required and optional fields for the WIF config in your `profiles.yml` file:
```

<SimpleTable>

| Field | Required | Description |
|-------|----------|-------------|
| `method` | Yes | Set to `external-oauth-wif`. |

> | `workload_pool_provider_path` | Yes | Full resource path of the GCP workload identity pool provider that trusts your Entra tenant, for example `//iam.googleapis.com/projects/123/locations/global/workloadIdentityPools/my-pool/providers/my-provider`. |

Reviewer 2: ```suggestion
| `workload_pool_provider_path` | Yes | The full resource path of the GCP workload identity pool provider that trusts your Entra tenant (for example, `//iam.googleapis.com/projects/123/locations/global/workloadIdentityPools/my-pool/providers/my-provider`). |
```

| `token_endpoint.type` | Yes | The external identity provider type. Only `entra` is currently supported. |

> | `token_endpoint.request_url` | Yes | The Entra OAuth token endpoint, for example `https://login.microsoftonline.com/<tenant>/oauth2/v2.0/token`. |

Reviewer 2: ```suggestion
| `token_endpoint.request_url` | Yes | The Entra OAuth token endpoint (for example, `https://login.microsoftonline.com/<tenant>/oauth2/v2.0/token`). |
```

| `token_endpoint.request_data` | Yes | The URL-encoded request body used to fetch the Entra token (for example `grant_type`, `client_id`, `client_secret`, and `scope`). |
| `service_account_impersonation_url` | No | The `generateAccessToken` URL of a service account to impersonate after federation, if you want BigQuery access scoped to that service account's permissions. |

> </SimpleTable>

Reviewer 2: Hey @mirnawong1 , the `workload_pool_provider_path` and `token_endpoint.request_url` rows currently include long sample values which forces horizontal scrolling when readers try to read the description column.

What do you think about removing the samples since they're both shown in the YAML code block example below? This way, readers wouldn't have to scroll to get the full description of each field.

Author: addressed

#### WIF config example in `profiles.yml`

> The following code snippet is an example of a WIF config in your `profiles.yml`, replacing the values with your own:

Reviewer 2: ```suggestion
The following code snippet is an example of a WIF config in your `profiles.yml`. Make sure to replace the values with your own:
```

For local development, we recommend using the OAuth method. If you're scheduling dbt on a server, you should use the service account auth method instead.

:::tip Workload Identity Federation

> WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/local/connect-data-platform/bigquery-setup?version=2.0&name=Fusion#supported-authentication-types). It's not supported in dbt Core v1.12 and earlier.

Reviewer 2: ```suggestion
WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/local/connect-data-platform/bigquery-setup?version=2.0&name=Fusion#supported-authentication-types). It's not supported in <Constant name="core" /> v1.12 and earlier.
```

> WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/fusion/connect-data-platform/bigquery-setup?version=2.0#workload-identity-federation-microsoft-entra). It's not supported in dbt Core v1.12 and earlier.

Author: added callout for versions 1.99 and lower

> WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/fusion/connect-data-platform/bigquery-setup?version=2.0#workload-identity-federation-microsoft-entra). It's not supported in dbt Core v1.12 and earlier.

Reviewer: this link is broken

https://docs-getdbt-com-git-udpate-add-bq-wif-dbt-labs.vercel.app/docs/fusion/connect-data-platform/bigquery-setup?version=2.0&name=Fusion#workload-identity-federation-microsoft-entra

> WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/fusion/connect-data-platform/bigquery-setup?version=2.0#workload-identity-federation-microsoft-entra). It's not supported in dbt Core v1.12 and earlier.

Author: ```suggestion
WIF authentication (`external-oauth-wif`) is available in [<Constant name="fusion" />](/docs/local/connect-data-platform/bigquery-setup?version=2.0&name=Fusion#supported-authentication-types). It's not supported in dbt Core v1.12 and earlier.
```
